main/src/ca/uwaterloo/flix/language/phase/jvm/ClosureInfo.scala

```diff
@@ -1,22 +1,22 @@
 package ca.uwaterloo.flix.language.phase.jvm
 
-import ca.uwaterloo.flix.language.ast.ErasedAst.FreeVar
+import ca.uwaterloo.flix.language.ast.ErasedAst.Expression
 import ca.uwaterloo.flix.language.ast.{MonoType, Symbol}
 
 /**
   * Meta information about a closure.
   */
-case class ClosureInfo(sym: Symbol.DefnSym, freeVars: List[FreeVar], tpe: MonoType) {
+case class ClosureInfo(sym: Symbol.DefnSym, closureArgs: List[Expression], tpe: MonoType) {
   /**
     * Returns the hash code of `this` closure info.
     */
-  override def hashCode(): Int = 7 * sym.hashCode + 11 * freeVars.hashCode()
+  override def hashCode(): Int = 7 * sym.hashCode + 11 * closureArgs.hashCode()
```

Comment:
@JonathanStarup 

I am concerned about this use of hash code on expressions. Why is equality needed on ClosureInfo? How is it used?

Computing hash code of expressions is not really well-defined...

Reply:
Only the types of the arguments were needed. the tpe was left out before so I'm not sure about how to compute equality. Would there ever be two different closures with the same symbol?